src/NzbDrone.Core/Parser/ParsingService.cs

```diff
@@ -354,91 +342,113 @@ private ParsedEpisodeInfo ParseSpecialEpisodeTitle(string title, Series series)
                 {
-                    possibleTitles.Add(altTitle.CleanSeriesTitle());
+                    return movieByImDb;
                 }
+            }
 
-                foreach (string title in possibleTitles)
+            if (searchCriteria != null)
+            {
+                Movie movieBySearchCriteria;
+                if (TryGetMovieBySearchCriteria(parsedMovieInfo, searchCriteria, out movieBySearchCriteria))
                 {
-                    if (title == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
-                    {
-                        possibleMovie = searchCriteria.Movie;
-                    }
+                    return movieBySearchCriteria;
+                }
+            }
 
-                    foreach (KeyValuePair<string, string> entry in romanNumeralsMapper)
-                    {
-                        string num = entry.Key;
-                        string roman = entry.Value.ToLower();
+            Movie movieByTitleAndOrYear;
+            if (TryGetMovieByTitleAndOrYear(parsedMovieInfo, out movieByTitleAndOrYear))
+            {
+                return movieByTitleAndOrYear;
+            }
 
-                        if (title.Replace(num, roman) == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
-                        {
-                            possibleMovie = searchCriteria.Movie;
-                        }
+            // nothing found up to here => logging that and returning null
+            _logger.Debug($"No matching movie {parsedMovieInfo.MovieTitle}");
+            return null;
+        }
 
-                        if (title.Replace(roman, num) == parsedMovieInfo.MovieTitle.CleanSeriesTitle())
-                        {
-                            possibleMovie = searchCriteria.Movie;
-                        }
-                    }
+        private bool TryGetMovieByImDbId(ParsedMovieInfo parsedMovieInfo, string imdbId, out Movie movie)
+        {
+            movie = _movieService.FindByImdbId(imdbId);
+            //Should fix practically all problems, where indexer is shite at adding correct imdbids to movies.
+            if (movie != null && parsedMovieInfo.Year > 1800 && parsedMovieInfo.Year != movie.Year)
+            {
+                movie = null;
+                return false;
+            }
+            return movie != null;
+        }
+
+        private bool TryGetMovieByTitleAndOrYear(ParsedMovieInfo parsedMovieInfo, out Movie movieByTitleAndOrYear)
+        {
+            Func<Movie, bool> isNotNull = movie => movie != null;
+            if (parsedMovieInfo.Year > 1800)
+            {
+                movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle, parsedMovieInfo.Year);
+                if (isNotNull(movieByTitleAndOrYear))
+                {
+                    return true;
                 }
+            }
+            movieByTitleAndOrYear = _movieService.FindByTitle(parsedMovieInfo.MovieTitle);
```

Comment:
This line is definitely called which would be the the previous `if-else-if` block (See Lines -406 through -418).